SafetyFirst/LazyList.fs

```diff
@@ -126,9 +126,23 @@ module  LazyList =
       | CellCons(a,b) -> consc a (append b l2)
 
     let delayed f = lzy(fun () ->  (getCell (f())))
-    let repeat x = 
-      let rec s = cons x (delayed (fun () -> s)) in s
 
+    let initInfinitely (initializer:NaturalInt -> 'a) = 
+      let rec s i () = consc (initializer i) (lzy (s i.Increment.AsNatural))
+      lzy (s NaturalInt.zero)
+
+    let initN count (initializer:NaturalInt -> 'a) =
+      let rec s i () = if i = count then CellEmpty else consc (initializer i) (lzy (s i.Increment.AsNatural))
+      lzy (s NaturalInt.zero)
+
+    let replicateInfinitely initial = 
+      let rec s () = consc initial (lzy s)
+      lzy s
```

Comment:
Is this the same thing as the old repeat function?  Why the change in implementation?